src/odemis/driver/hamamatsurx.py

```diff
@@ -625,15 +640,159 @@ def __init__(self, name, role, parent, daemon=None, **kwargs):
         timeRange = util.find_closest(timeRange, choices)  # make sure value is in choices
         self.timeRange = model.FloatEnumerated(timeRange, choices, setter=self._setTimeRange, unit="s")
```

Comment:
If I understand correctly, there is a difficult here, because the synchroscan unit doesn't provide a time range of the same unit as the single sweet unit. While the later one provides it in s, the former provides in "arbitrary values" 1,2,3,4... I've discussed with Herman, and apparently, for a given system, the arbitrary values can be fixed to a specific time. So, we could add an argument to the component, which allows to pass for each value the actual time range. Something like `time_ranges = {1: 2e-12, 2: 5e-12, 3: 8.9e-12 ...}`. It's only accepted if the time range has no unit. Then, we can have the VA always in s, and when it's used with the synchroscan, it's converted back and forth with the arbitrary values.